src/Planner/CollectColumnIndetifiers.h

```diff
@@ -0,0 +1,22 @@
+#pragma once
+
+#include <Analyzer/IQueryTreeNode.h>
+#include <Planner/Planner.h>
+#include <Planner/TableExpressionData.h>
+
+namespace DB
+{
+
+using ColumnIdentifierSet = std::unordered_set<ColumnIdentifier>;
+
+/// Collect all top level column identifiers from query tree node.
```

Comment:
Need to add in comment precondition that table expression data must be already collected in planner context.